Shark Cage/createACLinNewDesktop/createACLinNewDesktop/util.h

```diff
@@ -10,23 +10,39 @@
 #include <Windows.h>
 
 
-namespace util {
+namespace tokenLib {
 	/**
-	* This function is now obsolete, this approach is a dead end
 	* Function gets a SID of a group and creates a token with a group entry added in the token
-	* @param sid pointer to sid to be added to the token
-	* @param token reference to handle to requested token
+	* Source of the token is the token of the process itself. Returned token is identical to the calling process token, just includes one more group
+	* SE_CREATE_TOKEN_NAME must be held and enabled by a calling process, to successfully call this method
+	* Mind that there are security implications to the current implementations - such that f.x.:
+	* every process launched with token created by this function has SE_CREATE_TOKEN_NAME privilege, granting it basically any access to the system
+	* This can be mitigated by calling CreateRestrictedToken() function on the output of this method

```

Comment:
If I understand this correctly, for shark cage we should always call `CreateRestrictedToken` after using your lib function?

Reply:
Depends... if you are running a trusted code with that program it is not neccessary, but since the token will always hold `SE_CREATE_TOKEN_NAME` it will basically grant anything running under the token an unlimited access to the system (equivalent to yours). For user applications running in a cage, you should definately call `CreateRestrictedToken`... for stuff like CageLabeler it might not be necessary, but still a good practice